.github/workflows/service-build-test-upload.yml

```diff
@@ -105,7 +105,23 @@ jobs:
         run: |
           docker run --entrypoint "" ${{ env.IMAGE_NAME }} black --target-version py310 -S --check /usr/src/app
 
+      - name: Run isort
+        if: ${{ steps.changed-py-files.outputs.any_changed == 'true' && success() || failure() }}
+        run: |
+          docker run --entrypoint "" ${{ env.IMAGE_NAME }} isort --check-only --profile black ${{ steps.changed-py-files.outputs.all_changed_files }}
+
+      - name: Run flake8
+        if: ${{steps.changed-py-files.outputs.any_changed == 'true' && success() || failure() }}
+        run: |
+          docker run --entrypoint "" ${{ env.IMAGE_NAME }} flake8 ${{ steps.changed-py-files.outputs.all_changed_files }}
+
+      - name: Run mypy
+        if: ${{ steps.changed-py-files.outputs.any_changed == 'true' && success() || failure() }}
+        run: |
+          docker run --entrypoint "" ${{ env.IMAGE_NAME }} mypy ${{ steps.changed-py-files.outputs.all_changed_files }} --follow-imports=silent  --install-types --non-interactive
+
       - name: Run tests
+        if: success()
```

Comment:
question: Is this condition required?

The if condition is already `success()` by default if the Github documentation is correct, so I don't understand why do we need it.

Reply:
success() here is needed for indicating that tests should be ran only if linters succeeded